<bug>
          <bug_id>2529</bug_id>
          
          <creation_ts>2009-06-29 18:11:00 +0200</creation_ts>
          <short_desc>XmlHttpRequest problem in Safari 3 and Firefox 1.5</short_desc>
          <delta_ts>2014-03-03 15:25:44 +0100</delta_ts>
          <reporter_accessible>1</reporter_accessible>
          <cclist_accessible>1</cclist_accessible>
          <classification_id>1</classification_id>
          <classification>Unclassified</classification>
          <product>framework</product>
          <component>core.io</component>
          <version>master</version>
          <rep_platform>Safari 3</rep_platform>
          <op_sys>Windows</op_sys>
          <bug_status>CLOSED</bug_status>
          <resolution>FIXED</resolution>
          
          
          
          
          
          <priority>P4</priority>
          <bug_severity>minor</bug_severity>
          <target_milestone>0.8.3</target_milestone>
          
          
          
          <everconfirmed>1</everconfirmed>
          <reporter name="Daniel Wagner">daniel.wagner</reporter>
          <assigned_to name="Daniel Wagner">daniel.wagner</assigned_to>
          
          <qa_contact name="qooxdoo bugs mailing list">qooxdoo-bugs</qa_contact>
          <cf_browser>Safari</cf_browser>
          <cf_browser_version>3</cf_browser_version>

      

      

      

          <long_desc isprivate="0">
            <commentid>10666</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-06-29 18:11:32 +0200</bug_when>
            <thetext>The two tests in qx.test.io.remote.transport.XmlHttp fail in Safari 3 with 404 errors when trying to load the PHP scripts located in /framework/test/resource/qx/test/xmlhttp. The resourceManager correctly resolves the URIs into a path relative to the test application: ../../source/resource/qx/test/xmlhttp. 
However, the XmlHttpRequest sent by Safari goes to /source/resource/qx/test/xmlhttp. This would be correct if the request was sent from the parent window (the Test Runner) instead of the test application&#39;s Iframe, which is located one level below. This would suggest that Safari 3&#39;s XmlHttpRequest assumes it is running in the top frame&#39;s context. Then again, I could not reproduce this behavior by sending an identical request from a qooxdoo application loaded in a native browser iframe.

This was tested in Safari 3 on Windows, Safari 4/Win is unaffected, as is Chrome 2.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>10667</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-06-29 18:13:47 +0200</bug_when>
            <thetext>Set target milestone to 0.8.3</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>10733</commentid>
            <who name="Andreas Ecker">qooxdoo</who>
            <bug_when>2009-07-02 10:58:27 +0200</bug_when>
            <thetext>AFAIK this is the case for FF 1.5?</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>10744</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-07-02 15:15:46 +0200</bug_when>
            <thetext>Yes, Firefox 1.5 has the same bug. Changed the affected unit tests so they will fail in Safari 3/FF 1.5 if the request worked.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>11736</commentid>
            <who name="Andreas Ecker">qooxdoo</who>
            <bug_when>2009-09-01 11:19:27 +0200</bug_when>
            <thetext>reducing severity to minor, since this affects only legacy browsers in a rather unusual setup (iframes).</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>11737</commentid>
            <who name="Andreas Ecker">qooxdoo</who>
            <bug_when>2009-09-01 11:21:41 +0200</bug_when>
            <thetext>to Daniel, would you please comment about chances to fix this? To me this seems to be a browser-specific issue that probably isn&#39;t worth a dedicated workaround. I&#39;d suggest to close this as WONTFIX, leaving it as a know issue.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>11778</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-09-01 16:53:36 +0200</bug_when>
            <thetext>Agreed. I couldn&#39;t even reproduce this behavior outside of the Testrunner, so there&#39;s probably only a very small chance this will affect any real-life applications.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>13576</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-10-27 14:26:43 +0100</bug_when>
            <thetext>*** Bug 2932 has been marked as a duplicate of this bug. ***</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>13578</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-10-27 14:51:01 +0100</bug_when>
            <thetext>These tests have failed again in Safari 3 recently. In the Testrunner GUI, the test result was displayed twice, once as successful and once as failed due to the asynchronous test timeout being reached before resume() was called. After some investigation, it looks like the listener on the request&#39;s completed event was fired before the test even reached the wait() call. The listener callback calls resume(), which caused the GUI to display the test as successful. Then the wait() timeout expired, causing the test to be displayed again as failed. Wrapping the call to request.send() in a timeout so that wait() is called first seems to solve the problem.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>13580</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-10-27 14:52:44 +0100</bug_when>
            <thetext>Fixed in r20490 by adding a timeout before request.send(). This, however, does not explain the behavior described in the initial report.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>14284</commentid>
            <who name="Daniel Wagner">daniel.wagner</who>
            <bug_when>2009-11-23 10:57:03 +0100</bug_when>
            <thetext>The tests no longer fail in FF 1.5. Removed the inversion in r 20722</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>38856</commentid>
            <who name="Martin Wittemann">martin.wittemann</who>
            <bug_when>2014-03-03 15:25:44 +0100</bug_when>
            <thetext>Closed all bugs already shipped with a release.</thetext>
          </long_desc>
      
      

    </bug>